pkg/controller/rollout/rollout_canary.go

```diff
@@ -601,3 +644,36 @@ func (m *canaryReleaseManager) syncBatchRelease(br *v1beta1.BatchRelease, canary
 	}
 	return nil
 }
+
+// calculate next task
+func nextTask(reason string, currentTask v1beta1.FinalisingStepType) v1beta1.FinalisingStepType {
+	var taskSequence []v1beta1.FinalisingStepType
+	//REVIEW - should we consider more complex scenarios?
+	// like, user disables Rollout and rollbacks the workload at the same time?
+	switch reason {
+	case v1beta1.FinaliseReasonRollback: // rollback
+		taskSequence = []v1beta1.FinalisingStepType{
+			v1beta1.FinalisingStepTypeGateway,      // route all traffic to stable version
+			v1beta1.FinalisingStepTypeBatchRelease, // scale up old, scale down new
+			v1beta1.FinalisingStepTypeDeleteBR,
+			v1beta1.FinalisingStepTypeTrafficRouting, // do cleaning works(restore stable Service, remove canary Service)
```

Comment:
consider break the FinalisingStepTypeTrafficRouting step into two sub-steps: FinalisingStepTypeGateway and FinalisingStepTypeRestoreService